ams/parser.js

```diff
@@ -65,59 +65,50 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
     if (param.doi) {
       result.unitid = param.doi.split('/')[1];
     }
-  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null && param.tab_body == 'pdf') {
+  } else if ((match = /^\/(view|downloadpdf)\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.(xml|jpg)$/i.exec(path)) !== null) {
```

Comment:
There seems to be a DOI suffix in the URLs except for the JPEG, the DOI should be added in the scans, in the parser and the test file.

For example, this url https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=pdf contains the DOI suffix which is amsmonographs-d-18-0005.1 which matches the unitid you already added.

The DOI prefix seems to be 10.1175, you can create a constant to not copy it every time.

```js
const doiPrefix = '10.1175';
...
result.doi = `${doiPrefix}/${$result.unitid}`; // for example, expected for JPEG
```

Reply:
This doesn't seem like a good idea. It assumes that the same DOI prefix is used for all the content on this platform. The prefix identifies the registrant of the identifier this platform could conceivably content from multiple registrants, thus multiple prefixes.

Reply:
So far the pages I've looked at all seem to have the same DOI prefix, our practice is to create a constant.
If it turns out that there are other DOI prefixes, we can use a regex, but currently there is no reason to use a regex and not a constant.